### ISSUE TITLE
Uppercase fields appear on top in menus (sorting problem)

### ISSUE THREAD

**Issue by plauclair** [ryancramerdesign/ProcessWire#1310]
I don't normally create uppercase field names, but after a horrible refactoring accident on field creation code in the API this happened.. and I discovered this..

<img width="383" alt="capture d ecran 2015-07-27 a 15 38 31" src="https://cloud.githubusercontent.com/assets/171120/8915907/d2921fa4-3475-11e5-9f6b-9859376940a8.png">

**Comment by ryancramerdesign** [comment]
Not sure I understand. Field names can have any case (uppercase, lowercase, combination), according to what style you prefer. We recommend lowercase and underscore, since that's consistent with what will be used for the DB table names, but field names containing uppercase like in your screenshot is fine.

**Comment by teppokoivula** [comment]
Looks like a sorting quirk, actually. "P" shouldn't come before "b", regardless of case :)

**Comment by plauclair** [comment]
Yes, exactly. @ryancramerdesign it looks like it's using SORT_REGULAR.

```php
$words = array("i" => "I","l" => "love","P" => "ProcessWire");
ksort($words); // Implicit SORT_REGULAR
var_dump($words);



$words = array("i" => "I","l" => "love","P" => "ProcessWire");
ksort($words, SORT_NATURAL|SORT_FLAG_CASE);
var_dump($words);


// Or "better" yet, a MB-compatible approach
$words = array( 'love', 'ProcessWire', 'I');
$coll = new Collator('en_US');
$coll->sort( $words );
var_dump( $words );
```

Returns
```
array (size=3)
  'P' => string 'ProcessWire' (length=11)
  'i' => string 'I' (length=1)
  'l' => string 'love' (length=4)

array (size=3)
  'i' => string 'I' (length=1)
  'l' => string 'love' (length=4)
  'P' => string 'ProcessWire' (length=11)

array (size=3)
  0 => string 'I' (length=1)
  1 => string 'love' (length=4)
  2 => string 'ProcessWire' (length=11)
```

**Comment by teppokoivula** [comment]
It should be noted that both SORT_NATURAL and SORT_FLAG_CASE require PHP 5.4.0. Not sure if there's a way to do this without breaking things in 5.3.x.

**Comment by plauclair** [comment]
Oh, didn't realize that. Well a mix of usort and strcasecmp could also work. I actually prefer the Collator class myself but that would mean adding intl as a requirement if it isn't already..

**Comment by plauclair** [comment]
@ryancramerdesign you've marked this as not a bug, but I think it might be better labeled as an improvement at least. Feel free to close this if you don't think this should be fixed. Thanks!

**Comment by plauclair** [comment]
Fixed in #1442 and PR integrated.
